File: SmartG.Entities/Models/Page.cs
﻿using System;
using System.ComponentModel.DataAnnotations;
using System.Net.Mime;

namespace SmartG.Entities.Models
{
    public class Page:BaseEntity
    {
        public int PageId { get; set; }
        
        public string? Title { get; set; }
        public string? Content { get; set; }
        public string? Excerpt { get; set; }
        public string? MetaDescription { get; set; }
        public string? MetaKeyWords { get; set; }
        public string? Slug { get; set; }

        public int? SgImageId { get; set; }
        public int? SgGalleryId { get; set; }
        public ICollection<ContentBlock>? ContentBlocks { get; set; }
       
        


    }
}

File: SmartG-client/src/app/_interfaces/page.ts
import { ContentBlock } from "./content-block";
import { Gallery } from "./gallery";
import { Image } from "./image";

export interface Page {


        pageId: number;
        title: string;
        content: string;
        excerpt?: any;
        metaDescription: string;
        metaKeyWords: string;
        slug: string;
        image: Image;
        contentBlocks: ContentBlock[];
        dateCreated: Date;
        dateUpdated: Date;
        deleted: boolean;
        authorId?: any;
    gallery?:Gallery;
    sgImageId?: number;
    sgGalleryId?: any;
   
  

}
